lib/Sema/CSSimplify.cpp

```diff
@@ -3225,6 +3225,23 @@ bool ConstraintSystem::repairFailures(
     if (!isValueOfRawRepresentable(expectedType, rawReprType))
       return false;
 
+    auto valueTy = expectedType->lookThroughSingleOptionalType();
+    auto unwrappedRawReprTy = rawReprType->lookThroughSingleOptionalType();
+    auto memberLoc = getConstraintLocator(
+        locator.withPathElement(ConstraintLocator::ApplyFunction));
+    auto memberTy = createTypeVariable(memberLoc, TVO_CanBindToNoEscape);
+    addValueMemberConstraint(
+        MetatypeType::get(unwrappedRawReprTy, getASTContext()),
+        DeclNameRef(DeclName(getASTContext(), DeclBaseName::createConstructor(),
+                             {getASTContext().Id_rawValue})),
+        memberTy, DC, FunctionRefKind::SingleApply, {},
+        getConstraintLocator(
+            locator.withPathElement(ConstraintLocator::ConstructorMember)));
+    addConstraint(ConstraintKind::ApplicableFunction,
+                  FunctionType::get({AnyFunctionType::Param(valueTy)},
```

Comment:
I think this also needs a label `rawValue:` and that label should also be recorded via `associateArgumentLabels`

Reply:
It doesn't seem like it's accessible here, but I have done the equivalent of that for now.

Reply:
Ah, I thought that `associateArgumentLabels` is declared on constraint system...